dev/bots/analyze_sample_code.dart

```diff
@@ -439,9 +439,26 @@ class SampleChecker {
     if (_flutterVersion == null) {
       // Capture the flutter version information once so that the snippets tool doesn't
       // have to run it for every snippet.
+      if (verbose) {
+        print(<String>[_flutter, '--version', '--machine'].join(' '));
+      }
       final ProcessResult versionResult = Process.runSync(_flutter, <String>['--version', '--machine']);
+      if (verbose) {
+        stdout.write(versionResult.stdout);
+        stderr.write(versionResult.stderr);
+      }
       _flutterVersion = versionResult.stdout as String? ?? '';
     }
+    if (verbose) {
+      print(<String>[
+        Platform.resolvedExecutable,
+        'pub',
+        'global',
+        'run',
+        'snippets',
+        ...args,
+      ].join(' '));
+    }
     return Process.run(
```

Comment:
One thing here we could consider ahead of time is that things get stuck in `Process.run` (e.g. we don't correctly detect that the child process terminates) - if this is indeed the case then we are not going to see output from `Process.run`.

We might want to drop one level of abstraction here and use `Process.start` here (if verbose is passed) and forward `stdout` and `stderr` "eagerly" instead of waiting for completion. 

One more thing is that we could add an explicit timeout as well.